diff --git a/node_modules/goober/goober.d.ts b/node_modules/goober/goober.d.ts
index 91db638..dd87c20 100644
--- a/node_modules/goober/goober.d.ts
+++ b/node_modules/goober/goober.d.ts
@@ -70,7 +70,7 @@ declare namespace goober {
         tag: CSSAttribute | TemplateStringsArray | string,
         ...props: Array<string | number>
     ): string;
-    type StyledVNode<T> = (props: T, ...args: any[]) => any;
+    type StyledVNode<T> = ((props: T, ...args: any[]) => any) & { displayName?: string };
     type StylesGenerator<P extends Object = {}> = (props: P) => CSSAttribute | string;
     type Tagged<P extends Object = {}> = <PP extends Object = {}>(
         tag:
