diff --git a/node_modules/next/dist/client/index.js b/node_modules/next/dist/client/index.js
index 2713354..e0ced65 100644
--- a/node_modules/next/dist/client/index.js
+++ b/node_modules/next/dist/client/index.js
@@ -226,7 +226,7 @@ class Container extends _react.default.Component {
         if (!el) return;
         // If we call scrollIntoView() in here without a setTimeout
         // it won't scroll properly.
-        setTimeout(()=>el.scrollIntoView()
+        setTimeout(()=>el.scrollIntoView({ behavior: 'smooth' })
         , 0);
     }
     render() {
diff --git a/node_modules/next/dist/shared/lib/router/router.js b/node_modules/next/dist/shared/lib/router/router.js
index 845300e..e8d89cd 100644
--- a/node_modules/next/dist/shared/lib/router/router.js
+++ b/node_modules/next/dist/shared/lib/router/router.js
@@ -912,14 +912,14 @@ class Router {
         // First we check if the element by id is found
         const idEl = document.getElementById(hash);
         if (idEl) {
-            idEl.scrollIntoView();
+            idEl.scrollIntoView({ behavior: 'smooth' });
             return;
         }
         // If there's no element with the id, we check the `name` property
         // To mirror browsers
         const nameEl = document.getElementsByName(hash)[0];
         if (nameEl) {
-            nameEl.scrollIntoView();
+            nameEl.scrollIntoView({ behavior: 'smooth' });
         }
     }
     urlIsNew(asPath) {
